pkgdb/tests/lock-flake-installable.cc

```diff
@@ -0,0 +1,372 @@
+/* ========================================================================== *
+ *
+ *  @file lock-flake-installable.cc
+ *
+ *  @brief Tests for `lock-flake-installable` functionality.
+ *
+ * -------------------------------------------------------------------------- */
+
+#include "flox/lock-flake-installable.hh"
+#include "flox/core/util.hh"
+#include "test.hh"
+#include <fstream>
+#include <nix/flake/flake.hh>
+
+/* -------------------------------------------------------------------------- */
+
+
+/* -------------------------------------------------------------------------- */
+
+std::string
+unsupportedPackage( const std::string & system )
+{
+  if ( system == "aarch64-darwin" ) { return "glibc"; }
+  else if ( system == "x86_64-darwin" ) { return "glibc"; }
+  else if ( system == "aarch64-linux" ) { return "spacebar"; }
+  else if ( system == "x86_64-linux" ) { return "spacebar"; }
+  else
+    {
+      // Should be unreachable
+      return "wat?";
+    }
+}
+
+/**
+ * paths are relative to the test runner which in this case is the makefile in
+ * the pkgdb root
+ */
+const std::string localTestFlake
+  = std::filesystem::absolute( "./tests/data/lock-flake-installable" ).string();
+
+
+/* -------------------------------------------------------------------------- */
+
+bool
+test_attrpathUsesDefaults( const nix::ref<nix::EvalState> & state,
+                           const std::string &              system )
+{
+  auto lockedExplicit = flox::lockFlakeInstallable(
+    state,
+    system,
+    localTestFlake + "#packages." + system + ".hello" );
+
+  auto lockedImplicit
+    = flox::lockFlakeInstallable( state, system, localTestFlake + "#hello" );
+
+  EXPECT_EQ( nlohmann::json( lockedExplicit ),
+             nlohmann::json( lockedImplicit ) );
+
+  EXPECT_EQ( lockedImplicit.lockedFlakeAttrPath,
+             "packages." + system + ".hello" );
+
+  return true;
+}
+
+/**
+ * @brief Test that the flake origin is correctly parsed from the flake
```

Comment:
nonblocking: do you mean to make assertions about the locked flake installables? If not maybe change the comment to say that they are parsed (as opposed to correctly parsed)

Reply:
mainly trying different flakerefs and asserting (by means of it not throwing) that they parse.
I'll change the comment a bit to make that clearer